### DISCUSSION TITLE
Issue with mariadb Lock wait timeout exceeded; try restarting transaction

### DISCUSSION THREAD

**Discussion by rccess** [postalserver/postal#2926] [Help with using Postal]
i have Server with the following configuration
2xE5-2680v2 (2.8 GHz)
10 core
128 GB RAM
960 SSD

i am running non docker postal

postal configuration below

workers:
  quantity: 20
  threads: 20

main_db:
  host: 127.0.0.1
  username: xxxx
  password: xxxx
  database: migodb
  pool_size: 100

using Ip_pool with 20 IPs

------------------------
my mariadb configuration
in file /etc/mysql/mariadb.conf.d/50-server.cnf
[mysqld]
#
#
user                    = mysql
pid-file                = /run/mysqld/mysqld.pid
socket                  = /run/mysqld/mysqld.sock
#port                   = 3306
basedir                 = /usr
datadir                 = /var/lib/mysql
tmpdir                  = /tmp
lc-messages-dir         = /usr/share/mysql
#skip-external-locking

**bind-address            = 127.0.0.1
innodb_force_recovery = 3
innodb_buffer_pool_size = 3048M
max_connections         = 9000
connect_timeout         = 6
wait_timeout            = 31536000
max_allowed_packet      = 256M**

**but emails get stuck in queue after sending 20k emails and when i try to run the following command** 

**update queued_messages set locked_by = NULL, locked_at = NULL, retry_after = NULL, ip_address_id = NULL;**

get this error : **Lock wait timeout exceeded; try restarting transaction**

and no email sent, when check the log, same error showing in log

i have tried all the possible options which were highlighted in other threads but no luck, same issue again and again

please help me 

Thanks

**Comment by willpower232** [comment]
> i am running non docker postal

do you mean you are long overdue an update to Postal v3?

> Lock wait timeout exceeded; try restarting transaction

you need to have a look at the process list in your mariadb server for a clue as to what is keeping the table busy
